demos/text_detection_demo/cpp_gapi/src/custom_kernels.cpp

```diff
@@ -0,0 +1,408 @@
+                    points = imgPts;
+                    cv::resize(image, crop, outSize);
+                }
+            }
+            crop.reshape(1, blobShape).convertTo(blob, CV_32F);
+            out.emplace_back(blob.clone());
+            pts.push_back(points);
+        }
+    }
+};
+
+GAPI_OCV_KERNEL_ST(OCVCompositeTRDecode, custom::CompositeTRDecode, cv::GComputation) {
+    static void setup(const cv::GArrayDesc&, const cv::GArrayDesc&, const cv::gapi::GNetPackage&,
+                      const size_t, const size_t, std::shared_ptr<cv::GComputation>& graph) {
+        cv::GMat inDec, inHidden, feature, outHidden, outDec;
+        std::tie(outHidden, outDec) =
+            cv::gapi::infer<nets::TextRecognitionDecoding>(inDec, inHidden, feature);
+        graph = std::make_shared<cv::GComputation>(cv::GIn(inDec, inHidden, feature),
+                                                   cv::GOut(outHidden, outDec));
+    }
+    static void run(const std::vector<cv::Mat>&  hiddens_,
+                    const std::vector<cv::Mat>&  features,
+                    const cv::gapi::GNetPackage& net,
+                    const size_t                 numClasses,
+                    const size_t                 endToken,
+                          std::vector<cv::Mat>&  res,
+                          cv::GComputation& graph) {
+        constexpr int maxDecodedSymbols = 20;
+        GAPI_DbgAssert(hiddens_.size() == features.size());
+        const size_t numDetectedLabels = features.size();
+
+        if (0 == numDetectedLabels) {
+            res.clear();
+            return;
+        }
+
+        std::vector<cv::Mat> states(numDetectedLabels, cv::Mat({1}, CV_32FC1));
+        std::for_each(states.begin(), states.end(),
+                      [](cv::Mat& m){ m.dims = 1; *(m.begin<float>()) = 0; });
+        std::vector<cv::Mat> hiddens(hiddens_);
+
+        res.reserve(numDetectedLabels);
+        for (auto i = 0U; i < numDetectedLabels; i++) {
+            res.emplace_back(std::vector<int>{ maxDecodedSymbols, 1, static_cast<int>(numClasses) },
+                             CV_32FC1);
+        }
+
+        cv::Mat hidden, out;
+        for (size_t nLabel = 0; nLabel < numDetectedLabels; nLabel++) {
+            for (int nSymbol = 0; nSymbol < maxDecodedSymbols; nSymbol++) {
+                graph.apply(cv::gin(states[nLabel], hiddens[nLabel], features[nLabel]),
+                            cv::gout(hidden, out),
+                            cv::compile_args(net));
+
+                auto maxElem = std::max_element(out.begin<float>(),
+                                                out.begin<float>() + numClasses);
+                auto argmax = static_cast<size_t>(std::distance(out.begin<float>(), maxElem));
+                for (size_t i = 0; i < numClasses; i++) {
+                    res[nLabel].begin<float>()[nSymbol * numClasses + i] = out.begin<float>()[i];
+                }
+                if (endToken == argmax) {
```

Comment:
MB put this check after `auto argmax = static_cast...`.
Or `auto maxElem = std...` after check.

Reply:
No - `maxElem` is used for calculation `argmax` so it cannot be calculated after check.
And I cannot move the check upper - algorithm needs to write `out` down to `res` even if `endToken` has already been reached. Tried, doesn't work other way